### DISCUSSION TITLE
Support for user defined functions

### DISCUSSION THREAD

**Discussion by hantunes-uk** [autodiff/autodiff#225] [General]
Hi,

First of all superb library!
One thing I would like to know is if it is possible to add a custom functions (possibly with state) and if so how? I have 2 cases:

- first one is the Gauss cdf for which we have a "complex" approximation that is machine precision but for which we know the pdf analytically.  Note that in our implementation this Gauss class has 2 parameters (mean and std dev).
- second one is we use a 3rd party library for one dimensional interpolation. It is done via instantiation of an Interpolant class initialized early on. This class has 2 methods, one called value which computes the interpolated value for a given input x the other is derivatives which computes derivatives of the interpolator up to second order in x. 

The idea would for me to create some wrapper classes/functions that make these functions combine with autodiff. when computing derivatives. 

Thanks very much in advance.

**Comment by allanleal** [comment]
Sorry, it's difficult for me to give support and advice on very specific `autodiff` use cases. I hope you have figured out a solution by checking all examples. 
This example, https://autodiff.github.io/tutorials/#derivatives-of-a-multi-variable-function-with-parameters, may be helpful to you.
